src/runtime_src/core/tools/common/ReportCu.cpp

```diff
@@ -0,0 +1,147 @@
+
+  for (auto& line : custat) {
+    uint32_t ba = 0, cnt = 0, sta = 0;
+#ifdef _WIN32
+    sscanf_s(line.c_str(), "CU[@0x%x] : %d status : %d", &ba, &cnt, &sta);
+#else
+    sscanf(line.c_str(), "CU[@0x%x] : %d status : %d", &ba, &cnt, &sta);
+#endif
+
+    if (offset != (uint64_t)ba)
+      continue;
+
+    if (kind == cu_stat::usage)
+      ret = cnt;
+    else if (kind == cu_stat::stat)
+      ret = sta;
+
+    return ret;
+  }
+
+  return ret;
+}
+
+boost::property_tree::ptree
+populate_cus(const xrt_core::device *device, const std::string& desc)
+{
+  boost::property_tree::ptree pt;
+  std::vector<char> ip_buf;
+  std::vector<std::string> cu_stats;
+
+  pt.put("description", desc);
+ 
+  try {
+    ip_buf = xrt_core::device_query<qr::ip_layout_raw>(device);
+    cu_stats = xrt_core::device_query<qr::kds_custats>(device);
+  } catch (const std::exception& ex){
+    pt.put("error_msg", ex.what());
+  }
+
+  if(ip_buf.empty() || cu_stats.empty())
+    return pt;
+
+  const ip_layout *layout = (ip_layout *)ip_buf.data();
+  boost::property_tree::ptree ptCu_array;
+  for (int i = 0; i < layout->m_count; i++) {
+    if (layout->m_ip_data[i].m_type != IP_KERNEL)
+      continue;
+
+    uint32_t status = parseComputeUnitStat(cu_stats,
+			layout->m_ip_data[i].m_base_address, cu_stat::stat);
+    uint32_t usage = parseComputeUnitStat(cu_stats,
+			layout->m_ip_data[i].m_base_address, cu_stat::usage);
+    boost::property_tree::ptree ptCu;
+    std::string ss_addr = boost::str(boost::format("0x%x") %
+				     layout->m_ip_data[i].m_base_address);
+    ptCu.put( "name",         layout->m_ip_data[i].m_name);
+    ptCu.put( "base_address", ss_addr);
+    ptCu.put( "usage",        usage);
+    ptCu.put( "status",       xrt_core::utils::parse_cu_status(status));
+    ptCu_array.push_back(std::make_pair("", ptCu));
+  }
+
+  pt.add_child( std::string("compute_units"), ptCu_array);
+
+  return pt;
+}
+
+void
+ReportCu::getPropertyTreeInternal(const xrt_core::device * _pDevice, 
+                                              boost::property_tree::ptree &_pt) const
+{
+  // Defer to the 20202 format.  If we ever need to update JSON data, 
+  // Then update this method to do so.
+  getPropertyTree20202(_pDevice, _pt);
+}
+
+void 
+ReportCu::getPropertyTree20202( const xrt_core::device * _pDevice, 
+                                           boost::property_tree::ptree &_pt) const
+{
+  // There can only be 1 root node
+  _pt.add_child("Compute_Unit", populate_cus(_pDevice, "Compute Units Information"));
+}
+
+void 
+ReportCu::writeReport( const xrt_core::device * _pDevice,
+                                  const std::vector<std::string> & /*_elementsFilter*/, 
+                                  std::iostream & _output) const
+{
+  boost::property_tree::ptree _pt;
+  boost::property_tree::ptree empty_ptree;
+  getPropertyTreeInternal(_pDevice, _pt);
+
+  _output << boost::format("%s\n") % _pt.get<std::string>("Compute_Unit.description");

```

Comment:
What are you formatting here? boost::format() is doing nothing here.  Please remove it.